Sources/Sentry/SentrySubClassFinder.m

```diff
@@ -135,4 +120,55 @@ - (int)getClassList:(Class **)classes
     return numClasses;
 }
 
+- (BOOL)isClass:(Class)childClass subClassOf:(Class)parentClass
+{
+    if (!childClass || childClass == parentClass) {
+        return false;
+    }
+
+    // Using a do while loop, like pointed out in Cocoa with Love
+    // (https://www.cocoawithlove.com/2010/01/getting-subclasses-of-objective-c-class.html)
+    // can lead to EXC_I386_GPFLT which, stands for General Protection Fault and means we
+    // are doing something we shouldn't do. It's safer to use a regular while loop to check
+    // if superClass is valid.
+    while (childClass && childClass != parentClass) {
+        childClass = class_getSuperclass(childClass);
+    }
+
+    return childClass != nil;
+}
+
+- (void)actOnSubclassesOfViewControllerInImage:(NSString *)imageName block:(void (^)(Class))block;
+{
+    [self.dispatchQueue dispatchAsyncWithBlock:^{
+        unsigned int count = 0;
+        const char **classes = objc_copyClassNamesForImage(
+            [imageName cStringUsingEncoding:NSUTF8StringEncoding], &count);
+        Class viewControllerClass = NSClassFromString(@"UIViewController");
+        if (viewControllerClass == nil) {
+            [SentryLog logWithMessage:@"UIViewController class not found."
+                             andLevel:kSentryLevelDebug];
+            return;
+        }
+
+        NSMutableArray<NSString *> *classesToSwizzle = [NSMutableArray new];
+        for (int i = 0; i < count; i++) {
+            NSString *className = [NSString stringWithUTF8String:classes[i]];
+            if ([className hasSuffix:@"ViewController"]) {
+                Class class = NSClassFromString(className);
+                if ([self isClass:class subClassOf:viewControllerClass]) {
```

Comment:
`h`: We cant call this method on a background thread because of
https://github.com/getsentry/sentry-cocoa/blob/cb117ceaf5bb343a72635e3079269185a75b09b8/Sources/Sentry/SentrySubClassFinder.m#L61-L68

Maybe `objc_copyClassNamesForImage` is fast enough for running it on the main thread. If not we could just call `objc_copyClassNamesForImage` and then switch to the main thread.

Reply:
@philipphofmann  according to my tests, this solution does not trigger the class initializer.
The method `isClass:subClassOf:` is a local method, not a `Class` method.